### PR TITLE
Small changes in FineDoFHandlerView

### PR DESCRIPTION
**Pull request by peterrum**
- fix MPI type
- fix typo
- introduce a new tag

### CODE REVIEW

File: include/deal.II/base/mpi_tags.h [hunk 1]
```diff
@@ -142,6 +142,9 @@ namespace Utilities
           // Utilities::MPI::RemotePointEvaluation
           remote_point_evaluation,
 
+          // internal::FineDoFHandlerView::FineDoFHandlerView::reinit()
```
Comment (kronbichler) [inline include/deal.II/base/mpi_tags.h:145]:
```suggestion
          // internal::FineDoFHandlerView::FineDoFHandlerView::reinit() for mg global coarsening transfer
```

**Comment by peterrum** [comment]
FYI @nfehn

File: include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h [hunk 2]
```diff
@@ -658,13 +658,14 @@ namespace internal
...
-                      buffer.size(),
-                      Utilities::MPI::internal::mpi_type_id(buffer.data()),
-                      i.first,
-                      11,
-                      communicator,
-                      &requests.back());
+            MPI_Isend(
+              buffer.data(),
+              buffer.size(),
+              Utilities::MPI::internal::mpi_type_id(buffer.data()),
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h:664]: Can you please spell out the type of `buffer` in line 646, which is `std::vector<types::global_dof_index>`? This helps to see the data type more explicitly.

File: include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h [hunk 3]
```diff
@@ -709,20 +710,29 @@ namespace internal
...
-                     communicator,
-                     MPI_STATUS_IGNORE);
+            MPI_Recv(
+              buffer.data(),
+              buffer.size(),
+              Utilities::MPI::internal::mpi_type_id(buffer.data()),
+              status.MPI_SOURCE,
+              Utilities::MPI::internal::Tags::fine_dof_handler_view_reinit,
+              communicator,
+              MPI_STATUS_IGNORE);
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h:735]: Can you please add a check for the return value `ierr` of `MPI_Recv` (and possibly also in `MPI_Get_count` above), which might have revealed the mismatch in types for 64 bit integers?

**Review by kronbichler — COMMENTED** [review]
Good catch with the wrong count!

**Comment by peterrum** [comment]
@kronbichler Done!

**Review by kronbichler — APPROVED** [review]

**Comment by kronbichler** [comment]
Thanks!

**Comment by kronbichler** [comment]
I can confirm that this significantly improves performance with more than a few thousands of MPI ranks.

**Comment by drwells** [comment]
Should there be a changelog entry? We can put it in another PR but this seems like a notable fix.

**Pull request merged**